src/env-inl.h

```diff
@@ -321,6 +363,45 @@ inline void Environment::set_printed_error(bool value) {
   printed_error_ = value;
 }
 
+inline WorkerContext* Environment::worker_context() const {
+  CHECK(is_worker_instance());
+  CHECK_NE(worker_context_, nullptr);
+  return worker_context_;
+}
+
+inline void Environment::set_worker_context(WorkerContext* context) {
+  CHECK_EQ(worker_context_, nullptr);
+  CHECK_NE(context, nullptr);
+  worker_context_ = context;
+}
+
+inline Environment* Environment::owner_env() const {
+  CHECK_NE(owner_env_, nullptr);
+  return owner_env_;
+}
+
+inline void Environment::set_owner_env(Environment* env) {
+  CHECK_EQ(owner_env_, nullptr);
+  CHECK_NE(env, nullptr);
+  owner_env_ = env;
+}
+
+inline void Environment::set_instance_type(NodeInstanceType instance_type) {
+  instance_type_ = instance_type;
+}
+
+inline bool Environment::is_main_instance() const {
+  return instance_type_ == NodeInstanceType::MAIN;
+}
+
+inline bool Environment::is_worker_instance() const {
+  return instance_type_ == NodeInstanceType::WORKER;
+}
```

Comment:
it could check if worker_context_ is nullptr, but environment definitely needs a method to be able to tell this fact - it is needed in so many places.